web/modules/custom/usa_twig_vars/usa_twig_vars.module

```diff
@@ -46,82 +44,53 @@ function usa_twig_vars_preprocess(&$variables, $hook) {
+            case '/es/indice-agencias':
+              // Don't index beyond first page.
+              $current_letter = \Drupal::request()->query->get('letter') ?: 'a';
+              if ($current_letter != 'a') {
+                $noindex = TRUE;
+              }
+              break;
           }
 
-          $nodetype = $special_node_type ?: $node_real_type;
-          if ($nodetype == 'wizard_step') {
+          if ($node->getType() === 'wizard_step') {
             $noindex = TRUE;
           }
 
-          if (($node_real_type == 'directory_record') && $node->hasField('field_acronym')) {
+          if (($node->getType() === 'directory_record') && $node->hasField('field_acronym')) {
             if ($acronym = $node->get('field_acronym')->value) {
               $usa_twig_vars['head_title'] = $variables['head_title'];
               $usa_twig_vars['head_title']['title'] = Markup::create(trim(strip_tags($variables['page']['#title'])) .
                   " (" . trim(strip_tags($acronym)) . ")");
             }
           }
 
-          // Cleanup vars.
           $pageType = usa_twig_vars_get_page_type($node);
-          $usa_twig_vars['id'] = $node->id();
-          $usa_twig_vars['front'] = $isFront;
-          $usa_twig_vars['nodetype'] = $nodetype;
-          $usa_twig_vars['pagetype'] = $pageType;
-
-          $usa_twig_vars['meta_description'] = usa_twig_vars_get_meta_description($node);
-          $usa_twig_vars['meta_noindex'] = $noindex;
-
-          switch ($pageType) {
-            case 'Standard Page':
-              // If there's a special node type, use it.
-              $datalayer_Page_Type = $special_node_type ?: 'Content Page';
-              break;
-
-            case 'State Office Page':
-              // If there's a special node type, use it.
-              $datalayer_Page_Type = $special_node_type ?: $pageType;
-              break;
-
-            default:
-              $datalayer_Page_Type = $pageType ?? $nodetype;
-              break;
-          }
 
           // Data to be sent to GTM.
-          $usa_twig_vars['datalayer'] = usa_twig_vars_get_datalayer(
+          $datalayer = new TaxonomyDatalayerBuilder(
+            breadcrumbManager: \Drupal::service('breadcrumb'),
             node: $node,
             isFront: $isFront,
-            contentType: $nodetype,
             basicPagesubType: $pageType ?? NULL,
-            pageType: $datalayer_Page_Type,
           );
+
+          $usa_twig_vars['datalayer'] = $datalayer->build();
           unset($isFront);
+
+          // Cleanup vars.
+          // Promote variables figured out via datalayer builder that
+          // we initialized earlier.
+          // Need to get them from the array return after build() above.
+          $usa_twig_vars['id'] = $node->id();
+          $usa_twig_vars['nodetype'] = $usa_twig_vars['datalayer']['contentType'];
+          $usa_twig_vars['pagetype'] = $usa_twig_vars['datalayer']['basicPageSubtype'];
```

Comment:
I'm seeing a warning in my error log that 'basicPageSubtype' is an undefined array key. It's 'basicPagesubtype' on line 78.

Reply:
I would expect this to have a lot more effects than it does! page.html.twig has a use of it that affects the spacing under the breadcrumbs for "Navigation Cards Page" and I also see it in block--menu-block--navigation-page-items.html.twig but visually I don't see the effect. 

Anyway, just fix it so twig still has "pagetype" defined and all will be well.

Reply:
I've fixed the typo and then realized I didn't have Drupal configured to show errors and warning on screen after my last DB sync. Grrr.

In theory, at the higher levels, this is something PHPStan could catch once you describe array shapes.